src/java.base/share/classes/java/text/CompactNumberFormat.java

```diff
@@ -47,112 +47,119 @@
  * range, or considered as a default pattern for an empty string.
  *
- * <p>
+ * <h3>Negative Subpatterns</h3>
  * A compact pattern contains a positive and negative subpattern
- * separated by a subpattern boundary character {@code ';' (U+003B)},
+ * separated by a subpattern boundary character {@code ';'},
  * for example, {@code "0K;-0K"}. Each subpattern has a prefix,
  * minimum integer digits, and suffix. The negative subpattern
  * is optional, if absent, then the positive subpattern prefixed with the
- * minus sign ({@code '-' U+002D HYPHEN-MINUS}) is used as the negative
+ * minus sign ({@code '-'}) is used as the negative
  * subpattern. That is, {@code "0K"} alone is equivalent to {@code "0K;-0K"}.
  * If there is an explicit negative subpattern, it serves only to specify
  * the negative prefix and suffix. The number of minimum integer digits,
  * and other characteristics are all the same as the positive pattern.
  * That means that {@code "0K;-00K"} produces precisely the same behavior
  * as {@code "0K;-0K"}.
  *
- * <p>
+ * <h4>Escaping Special Characters</h4>
  * Many characters in a compact pattern are taken literally, they are matched
  * during parsing and output unchanged during formatting.
  * {@linkplain DecimalFormat##special_pattern_character Special characters},
  * on the other hand, stand for other characters, strings, or classes of
- * characters. They must be quoted, using single quote {@code ' (U+0027)}
+ * characters. These characters must be quoted using single quotes ({@code '})
  * unless noted otherwise, if they are to appear in the prefix or suffix
  * as literals. For example, 0\u0915'.'.
  *
  * <h3>Plurals</h3>
  * <p>
- * In case some localization requires compact number patterns to be different for
- * plurals, each singular and plural pattern can be enumerated within a pair of
- * curly brackets <code>'{' (U+007B)</code> and <code>'}' (U+007D)</code>, separated
- * by a space {@code ' ' (U+0020)}. If this format is used, each pattern needs to be
- * prepended by its {@code count}, followed by a single colon {@code ':' (U+003A)}.
- * If the pattern includes spaces literally, they must be quoted.
+ * Plural rules differ between locales, in some cases, the compact number patterns
+ * will have distinct forms for singular and plural.
  * <p>
- * For example, the compact number pattern representing millions in German locale can be
- * specified as {@code "{one:0' 'Million other:0' 'Millionen}"}. The {@code count}
- * follows LDML's
- * <a href="https://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules">
- * Language Plural Rules</a>.
+ * For example, while the pattern representing millions ({@code 10}<sup>{@code 6}
+ * </sup>) in the US locale can be specified as the SimplePattern: {@code "0 Million"}, for the
+ * German locale it can be specified as the PluralPattern:
+ * {@code "{one:0' 'Million other:0' 'Millionen}"}.
```

Comment:
I like the German example, however, I think the original explanation for the pattern should be preserved (wording might be better though)

Reply:
I originally removed it because the syntax already covered proper usage. But I can see how it can be confusing at first to understand. I re-inserted the original wording, but modified it IMO to be easier to understand.